riscv/mmu.h

```diff
@@ -100,6 +100,11 @@ class mmu_t
   // template for functions that load an aligned value from memory
   #define load_func(type, prefix, xlate_flags) \
     type##_t ALWAYS_INLINE prefix##_##type(reg_t addr, bool require_alignment = false) { \
+      if (!matched_trigger) { \
+        matched_trigger = trigger_exception(triggers::OPERATION_LOAD, addr); \
+        if (matched_trigger) \
+          throw *matched_trigger; \
+      } \
```

Comment:
This needs to be done in such a way as to not slow down the common case.  For a test that is heavy in loads and stores, it results in a 200% increase in runtime, since it calls `trigger_exception` on every load and store!

We need to handle the common case (aligned + TLB hit) with no additional overhead.  It should even be possible to do without any additional `if` statements by reordering the code.